documentation/modules/ROOT/pages/integrations/cloudevents.adoc

```diff
@@ -167,6 +167,90 @@ The following example shows how to configure the CloudEvents converter to emit c
 The CloudEvents converter converts Kafka record values. In the same connector configuration, you can specify `key.converter` if you want to operate on record keys.
 For example, you might specify `StringConverter`, `LongConverter`, `JsonConverter`, or `AvroConverter`.
 
+// Type: concept
+// ModuleID: configuration-of-sources-of-metadata-and-some-cloudevents-fields
+// Title: Configuration of sources of metadata and some CloudEvents fields
+[[configuration-of-sources-of-metadata-and-some-cloudevents-fields]]
+== Configuration of sources of metadata and some CloudEvents fields
+
+By default, the `metadata.location` property consists of three parts, as seen in the following example:
+
+[source,json,indent=0,subs="+attributes"]
+----
+"value,id:generate,type:generate"
+----
+
+The first part specifies the source for retrieving a record's metadata; the permitted values are `value` and `header`. 
+The next parts specify how to obtain the `id` and `type` fields of a CloudEvent; the permitted values are `generate` and `header`.
+
+=== Obtaining record metadata
+
+To construct a CloudEvent, the converter requires source, operation, and transaction metadata.
+Generally, the converter can retrieve the metadata from a record's value. 
+But in some cases, before the converter receives a record, the record might be processed in such a way that metadata is not present in its value, for example, after the record is processed by the Outbox Event Router SMT.
+To preserve the required metadata, you can use the following approach to pass the metadata in the record headers.
+.Procedure
+
+1. When the record is processed by other pipeline jobs before it reaches the converter, implement a mechanism for recording the metadata in the message headers, for example, by using the `HeaderFrom` SMT.
```

Comment:
I think, this need to change somehow because a user should add metadata to headers whether or not there are the jobs before the converter (if the user enabled 2.). Even if there is no one job before the converter, probably the user should create one (for example, HeadersFrom transform) just to add headers